api/services/AccessTokenAuth.js

```diff
@@ -0,0 +1,35 @@
+const crypto = require('crypto')
+const Promise = require('bluebird')
+
+var AccessTokenAuth = module.exports = {
+
+  generateToken: function() {
+    const randomBytes = Promise.promisify(crypto.randomBytes)
+    return randomBytes(24).then(buffer => buffer.toString('hex'))
+  },
+
+  checkAndSetAuthenticated: function(req) {
+    req.body = req.body || {}
+    const token = req.body.access_token || req.query.access_token || req.headers['x-access-token']
+    if (!token) return Promise.resolve()
+    return User.query(function (qb) {
```

Comment:
check out the implementation of `UserSession.login` and maybe add a new argument if you don't want API token use to increment `last_login` on the user